engine/renderer/public/pipelines/fxaa_pipeline.hpp

```diff
@@ -0,0 +1,55 @@
+﻿#pragma once
+
+#include "frame_graph.hpp"
+#include "gbuffers.hpp"
+
+#include <cstdint>
+#include <memory>
+
+class BloomSettings;
+class GraphicsContext;
+class CameraResource;
+struct GPUImage;
+struct RenderSceneDescription;
+
+class FXAAPipeline final : public FrameGraphRenderPass
+{
+public:
+    FXAAPipeline(const std::shared_ptr<GraphicsContext>& context, const GBuffers& gBuffers, const ResourceHandle<GPUImage>& fxaaTarget, const ResourceHandle<GPUImage>& sourceTarget);
+    ~FXAAPipeline() final;
+
+    void RecordCommands(vk::CommandBuffer commandBuffer, uint32_t currentFrame, const RenderSceneDescription& scene) final;
+
+    float& GetEdgeTreshholdMin() { return _pushConstants.edgeThresholdMin; }
+    float& GetEdgeTreshholdMax() { return _pushConstants.edgeThresholdMax; }
+    float& GetSubPixelQuality() { return _pushConstants.subPixelQuality; }
+    int32_t& GetIterations() { return _pushConstants.iterations; }
+    bool& GetEnableFXAA() { return _pushConstants.enableFXAA; }
+
```

Comment:
Don't write getter functions like this, either make the variables public or add the classes that need access friend classes

Reply:
I dont know about this one. Nobody said anything when I did the same thing in the SSAO pr. If its a big problem i can change it. The rest of the things are pushed now 🫡

Reply:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-get 
Its fine, but you should read this sometime